### PR TITLE
[vividus] Optimize stories search for JUnit runner

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus/src/main/java/org/vividus/runner/GenericRunner.java [hunk 1]
```diff
@@ -51,6 +52,8 @@
     private final Embedder embedder;
     private final IBatchedPathFinder batchedPathFinder;
 
+    private Map<String, List<String>> batchToStoryPaths;
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/runner/GenericRunner.java:55]: this logic should be moved to `BatchedPathFinder`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus/src/main/java/org/vividus/runner/GenericRunner.java (hunk 1 above)
Comment (uarlouski) [reply vividus/src/main/java/org/vividus/runner/GenericRunner.java:55]: Why?  It still won't be clear from the main logic that this value gets cached, so each time someone gets to this point he will have to lookup BatchedPathFinder class

File: vividus-engine/src/main/java/org/vividus/BatchedPathFinder.java [hunk 2]
```diff
@@ -25,25 +25,35 @@
 import org.vividus.batch.BatchResourceConfiguration;
 import org.vividus.batch.BatchStorage;
 
+/**
+ * The API provided by this class <b><u>is not</u></b> thread-safe and expected to be invoked from the main thread
```
Comment (valfirst) [inline vividus-engine/src/main/java/org/vividus/BatchedPathFinder.java:29]:
```suggestion
 * The API provided by this class is <b><u>not</u></b> thread-safe and expected to be invoked from the main thread
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

File: vividus-engine/src/test/java/org/vividus/BatchedPathFinderTests.java [hunk 3]
```diff
@@ -54,8 +50,9 @@ void testFindPaths() throws IOException
         when(batchStorage.getBatchResourceConfigurations()).thenReturn(Map.of(batchKey, batchResourceConfiguration));
         List<String> testPaths = List.of("testPath");
         when(pathFinder.findPaths(batchResourceConfiguration)).thenReturn(testPaths);
-        Map<String, List<String>> actual = batchedPathFinder.findPaths();
         Map<String, List<String>> expected = Map.of(batchKey, testPaths);
-        assertEquals(expected, actual);
+        assertEquals(expected, batchedPathFinder.getPaths());
```
Comment (ikalinin1) [inline vividus-engine/src/test/java/org/vividus/BatchedPathFinderTests.java:54]: how caching is tested? Shouldn't you verify invocation quantity?

Comment (uarlouski) [reply vividus-engine/src/test/java/org/vividus/BatchedPathFinderTests.java:54]: please check verifyNoMoreInteractions below

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**